promql/engine.go

```diff
@@ -3844,3 +3982,75 @@ func (ev *evaluator) gatherVector(ts int64, input Matrix, output Vector, bufHelp
 
 	return output, bufHelpers
 }
+
+// timeValueOf calculates the scalar time value of an expression in float64.
```

Comment:
Currently, it's more a duration than a time value. I guess later we want to use the same for timestamps (with `@`), so maybe `timeValue` is a good word, but this should explain that it might stand for a duration in seconds or a timestamp in seconds.